config/routes.rb

```diff
@@ -1,9 +1,21 @@
 Autolab3::Application.routes.draw do
-  root "courses#index"
 
   devise_for :users, controllers: { omniauth_callbacks: "users/omniauth_callbacks" },
                      path_prefix: "auth"
 
+  devise_scope :user do
+
+    authenticated :user do
+      root 'courses#index', as: :authenticated_root
+    end
+
+    unauthenticated do
+      root 'devise/sessions#new', as: :unauthenticated_root
+    end
+
+  end
```

Comment:
What is this block doing?  It seems to create 2 different routes that have the same path.  The result is that we get two different route helper functions pointing to the same route, but I think anyone who hits "/" will always be directed to only one of the actions.